smartsim/_core/_install/builder.py

```diff
@@ -278,17 +296,21 @@ def build_from_git(
         if not self.is_valid_url(git_url):
             raise BuildError(f"Malformed {database_name} URL: {git_url}")
 
+        clone_cmd = [self.binary_path("git"), "clone"]
+        if self._platform == (OperatingSystem.DARWIN, Architecture.ARM64):
```

Comment:
Should we move this into a method to avoid repeating it and maybe clarify what's happening?

```py
def augment_git_cmd(platform, cmd: t.List[str]):
   """Modify git commands to include autocrlf when on a platform that needs
   autocrlf enabled to behave correctly"""
   if platform == (OperatingSystem.DARWIN, Architecture.ARM64):
      where = cmd.index("git") + 1
      return cmd[:where] + ["--config", "core.autocrlf=true"] + cmd[:where]
   return cmd
```

Reply:
I like the suggestion